.github/workflows/rust.yml

```diff
@@ -10,51 +10,163 @@ env:
-      - name: Build
-        run: cargo build
-      - name: Build documentation
-        run: cargo doc --no-deps --quiet
-      - name: Code analysis
-        run: cargo clippy --quiet
-      - name: Run tests with JSON (default) codec
-        run: cargo test --quiet
-      - name: Run tests with Bincode codec
-        run: cargo test --no-default-features --quiet --features full --features default-codec-bincode
-      - name: Run tests with CBOR (serde_cbor) codec
-        run: RUSTFLAGS="-A deprecated" cargo test --no-default-features --quiet --features full --features default-codec-cbor
-      - name: Run tests with CBOR (ciborium) codec
-        run: cargo test --no-default-features --quiet --features full --features default-codec-ciborium
-      - name: Run tests with MessagePack codec
-        run: cargo test --no-default-features --quiet --features full --features default-codec-message-pack
-      - name: Test rch feature only
-        run: cargo test --no-default-features --quiet --features rch --features default-codec-json
-      - name: Test rfn feature only
-        run: cargo test --no-default-features --quiet --features rfn --features default-codec-json
-      - name: Test robj feature only
-        run: cargo test --no-default-features --quiet --features robj --features default-codec-json
-      - name: Test rtc feature only
-        run: cargo test --no-default-features --quiet --features rtc --features default-codec-json
-      - name: Test with MSRV 1.51
-        run: |
-          rustup install 1.51
-          cargo +1.51 test --quiet
-      - name: Install Cargo Tarpaulin for code coverage
-        run: cargo install cargo-tarpaulin --locked --quiet
-        continue-on-error: true
-      - name: Code coverage
+
+      - name: Install Rust ${{ env.minrust }}
+        run: rustup default ${{ env.minrust }}
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Run cargo check
+        run: cargo check
+  
+  # Lints
+  build-docs:
+    name: Build docs
+    needs: [msrv]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Build docs
+        env:
+          RUSTDOCFLAGS: -D rustdoc::broken_intra_doc_links
+        run: cargo doc --no-deps
+  
+  clippy:
+    name: Clippy
+    needs: [msrv]
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Cache dependencies
+        uses: Swatinem/rust-cache@v1
+
+      - name: Run clippy
+        env:
+          RUSTFLAGS: -D warnings
+        run: cargo clippy --tests
+  
+  rustfmt:
+    name: Format
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v2
+
+      - name: Run cargo fmt
+        run: cargo fmt --all -- --check
+
+  # Tests
+  test:
+    name: Test
+    needs: [msrv]
```

Comment:
I don't understand why this depends on `msrv`. Are the tests run with current stable Rust or with MSRV Rust?

Reply:
All jobs are run in a fresh environment, so all jobs except `msrv` uses the current stable Rust (included in the base ubuntu-latest image). Setting a different default version in `msrv` doesn't affect other jobs, even if they depend on `msrv`.